src/plugins/hidsim/gui/vmi_win_gui_utils.h

```diff
@@ -0,0 +1,220 @@
+ ***************************************************************************/
+
+#ifndef VMI_WIN_GUI_UTILS_H
+#define VMI_WIN_GUI_UTILS_H
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <wchar.h>
+#include <getopt.h>
+
+#include <libvmi/libvmi.h>
+
+#include "vmi_win_gui_offsets.h"
+
+extern struct Offsets symbol_offsets;
+
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+
+/*
+ * Checks, whether the _OBJECT_HEADER, which precedes every executive object,
+ * is preceded by an optional header of type _OBJECT_HEADER_NAME_INFO.
+ * If this is the case, the name of the executive object is read and returned.
+ *
+ */
+char* retrieve_objhdr_name(vmi_instance_t vmi, addr_t addr)
+{
+    addr_t obj_hdr = 0;
+    addr_t obj_hdr_nameinfo_addr= 0;
+    uint8_t im = 0;
+    char* name = NULL;
+    unicode_string_t* us = NULL;
+    unicode_string_t out = { .contents = NULL };
+
+    /*
+     * Retrieves the beginning of the _OBJECT_HEADER, to find it subtract the
+     * offset to the body from current address. Since the current executive
+     * object is _partly_ incorporated in the size of the _OBJECT_HEADER-struct
+     *
+     * See "The Art of Memory Forensics", p. 119 ff. for a description
+     */
+    obj_hdr = addr - symbol_offsets.objhdr_body_offset;
+    obj_hdr_nameinfo_addr = obj_hdr;
+
+    if (VMI_FAILURE == vmi_read_8_va(vmi, obj_hdr + symbol_offsets.objhdr_infomask_offset,
+            0, &im))
+    {
+        fprintf(stderr, "Error reading InfoMask from _OBJECT_HEADER at: %" PRIx64
+            "\n", obj_hdr);
+        return NULL;
+    }
+
+    /*
+     * Checks, if there comes an optional _OBJECT_HEADER_CREATOR_INFO after
+     * _OBJECT_HEADER_NAME_INFO, which has to added to the offset to subtract
+     * from the address signifying the start of the _OBJECT_HEADER
+     */
+    if (im & OBJ_HDR_INFOMASK_CREATOR_INFO)
+        obj_hdr_nameinfo_addr -= symbol_offsets.objhdr_creator_info_length;
+
+    /* Returns NULL immediately, if there is no _OBJECT_HEADER_NAME_INFO */
+    if (!(im & OBJ_HDR_INFOMASK_NAME))
+        return NULL;
+
+    obj_hdr_nameinfo_addr -= symbol_offsets.objhdr_name_info_length;
+
+    us = vmi_read_unicode_str_va(vmi, obj_hdr_nameinfo_addr +
+            symbol_offsets.objhdr_name_info_name_offset, 0);
+
+    if (us && VMI_SUCCESS == vmi_convert_str_encoding(us, &out, "UTF-8"))
+    {
+        name = strndup((char*) out.contents, out.length);
+        free(out.contents);
+    }
+
+    if (us)
+        vmi_free_unicode_str(us);
+
+    return name;
+}
+
+/*
+ * Reads a Windows wchar-string into a wchar_t*, since vmi_read_unicode_str_va
+ * fails to parse _RTL_ATOM_ENTRY's name-string or _LARGE_UNICODE_STRINGs.
+ * Expansion is performed since Windows' wchar is 2 bytes versus 4 bytes on
+ * 64bit-Linux
+ */
+wchar_t* read_wchar_str_pid(vmi_instance_t vmi, addr_t start, size_t len, vmi_pid_t pid)
+{
+    wchar_t* s = (wchar_t*) malloc(sizeof(wchar_t) * len);
```

Comment:
Maybe I'm a bore but.. you never check malloc returning value. Null pointer dereference is UB and that's not good idea as for me. (this is not the only place in your code without this check)

Reply:
Thanks for your review, @exescript. 
Should I check `malloc`s success each and every time I call the function in hidsim or just in there?
 
Should I maybe write a wrapper function like this
```c
void * xmalloc(size_t size) 
{
    void* p; 
    p = malloc(size); 
    if (!p)
    {
        PRINT_DEBUG(...);
        exit(1);
    }  
    return p;
}
```
which is called throughout my code?

Reply:
Don't ask me as I'd tell you to purge out of existence every instance of `malloc`/`new` ¯\\\_(ツ)\_/¯

Reply:
Personally, I would avoid using libc functions such as malloc/free in C++ code.

Reply:
> Personally, I would avoid using libc functions such as malloc/free in C++ code. But yeah, checking for allocation failures would be nice.

Well, I implemented most of the plugin in raw C-style, which I voted with @BonusPlay and @icedevml... 
I will add checks for allocation failures.

Reply:
I think that the simplest way would be to just use glib allocator functions - g_malloc.
We're already linking against it so IMO it would be better than writing custom wrappers.

Reply:
> I think that the simplest way would be to just use glib allocator functions such as g_malloc.

`g_malloc` will abort the application, when the memory application fails (see e.g. [here](http://www.gtk-server.org/gtk1/glib/glib-memory-allocation.html)). Since this is not what should be done for a thread, which is not absolutely mission critical for DRAKVUF's overall goal, I decided to stay with glibc's `malloc` and added the appropriate checks. Those were added in 467a35a4dd03721782051369be2699d463a37fcf. Thanks for the hint, @exescript, and the corresponding input, @chivay!